src/oic/oic/provider.py

```diff
@@ -1202,7 +1202,9 @@ def _do_user_info(self, token, **kwargs):
         info = self.schema(**self._collect_user_info(session))
 
         # Should I return a JSON or a JWT ?
-        _cinfo = self.cdb[session["client_id"]]
+        _cinfo = self.cdb.get(session["client_id"])
+        if _cinfo is None:
+            return error_response('unauthorized_client', descr='Unknown client')
```

Comment:
This could be handled inside the **except** block. Isn't?

Reply:
Yes, but not all implementations of ClientDatabase raise `KeyError`, so I went with safer approach.